v2/report_publisher.go

```diff
@@ -1,27 +1,106 @@
 package bugsnag
 
-import "fmt"
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+)
 
 type reportPublisher interface {
 	publishReport(*payload) error
+	setMainProgramContext(context.Context)
 }
 
-type defaultReportPublisher struct{}
+var mainProgramDone = false
```

Comment:
This is only used within `defaultReportPublisher`, so it should be part of the struct. Otherwise multiple publishers would fight over it.

Perhaps give it a more generic name like `isClosed` so that you could close the report publisher for different reasons.